crossopt/assignment1/Q2.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+
+
+class Node:
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+
+
+class LinkedList:
+    def __init__(self):
+        self.head = None
+
+    # adds element to front of list
+    def insert(self, new_elem):
+        new_node = Node(new_elem)
+        new_node.next = self.head
+        self.head = new_node
+
+    def length(self):
+        ans = 0
+        cnt = self.head
+        while cnt:
+            ans += 1
+            cnt = cnt.next
+        return ans
+
+    def __str__(self):
+        ans = []
+        cnt = self.head
+        while cnt:
+            ans.append(str(cnt.data))
+            cnt = cnt.next
+        return "{" + ", ".join(ans) + "}"
+
+    # return kth element from end of list
+    def getKth(self, k):
```

Comment:
Do you think you can make the bracket notation work?
This means means that your container linkedlist would work with the syntax:
llist = LinkedList()
llist.insert(5)
llist.insert('dog')
llist.insert(set([1, 2, 3]))
llist[-1], llist[-2], llist[-3]

If you use the Pythonic convention like bracket for accessing an element by its index, the benefit is for your users: they can quickly learn how to use your data structure without a lot of effort, because the syntax is the same as for most other container like dictionary or lists.